src/qmflows/settings.py

```diff
@@ -96,7 +102,59 @@ def update(self, other):
                     pred1 = (br in other[name] and other[name][br])
                     pred2 = (br in self[name] and self[name][br])
                     if pred1 or pred2:
-                        self[name] = Settings()
+                        self[name] = cls()
                     self[name].update(other[name])
             else:
                 self[name] = other[name]
+
+
+def _unsuported_operation(meth: Callable) -> Callable[..., NoReturn]:
+    """Decorate a method such that it raises a :exc:`TypeError`."""
+    @wraps(meth)
+    def new_meth(self, *args, **kwargs) -> NoReturn:
+        raise TypeError(f"{self.__class__.__name__!r} does not "
+                        "support item assignment or deletion")
+    return new_meth
+
+
+def _super_method(meth: Callable[..., '_Settings']) -> Callable[..., '_Settings']:
```

Comment:
If I understand it well, this method is a wrapper to call the parent method, right? shouldn't that be handle by the class itself?

Reply:
More specifically, it calls the it's designed to call the method of a super-class but then return an instance of the sub-class.